src/aleph/vm/utils/__init__.py

```diff
@@ -75,7 +75,7 @@ def to_json(o: Any):
     elif hasattr(o, "dict"):  # Pydantic
         return o.dict()
     elif is_dataclass(o):
-        return dataclass_as_dict(o)
+        return dataclass_as_dict(o)  # type: ignore
```

Comment:
What is the error and isn't there a more elegant way to solve the issue ?
`# type: ignore` looks a bit like "force-ignore-this".

Reply:
And shouldn't the return type be defined for this function?

Reply:
@hoh That's explained in the PR description why I had to force ignore.

error was
```
src/aleph/vm/utils/__init__.py:78: error: No overload variant of "asdict" matches argument type "type[DataclassInstance]"  [call-overload]
src/aleph/vm/utils/__init__.py:78: note: Possible overload variants:
src/aleph/vm/utils/__init__.py:78: note:     def asdict(obj: DataclassInstance) -> dict[str, Any]
src/aleph/vm/utils/__init__.py:78: note:     def [_T] asdict(obj: DataclassInstance, *, dict_factory: Callable[[list[tuple[str, Any]]], _T]) -> _T
Found 1 error in 1 file (checked 71 source files)
```
DataclassInstance is not something that exist, so for me it looks like a mypy bug.  But if you have another fix I would be happy to accept it.

@Psycojoker which function?

Reply:
@olethanh github doesn't allow me to comment on any line but the exact function you are editing

![image](https://github.com/user-attachments/assets/96ca4d28-28d2-4999-b8bb-d7bf5169a9b6)

Reply:
yeah it probably should but it's totally outside the scope of this PR

Reply:
Ended up doing it here https://github.com/aleph-im/aleph-vm/pull/704

Reply:
I merged #704